File: benchmarkDS/Vanilla/GEMINI_gemini-2.5-pro/Lse21.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <errno.h>

/**
 * @brief Adds 1000 to the given number.
 * 
 * @param number The integer to which 1000 will be added.
 * @return The sum of the number and 1000.
 */
int addThousand(int number) {
    return number + 1000;
}

/**
 * @brief Main function. If a command-line argument is provided, it adds 1000 to it and prints the result.
 * Otherwise, it runs 5 hardcoded test cases.
 * 
 * @param argc The number of command-line arguments.
 * @param argv An array of command-line arguments.
 * @return 0 on success, 1 on error.
 */
int main(int argc, char *argv[]) {
    if (argc > 1) {
        char *endptr;
        errno = 0; // To distinguish success/failure after call
        long inputVal = strtol(argv[1], &endptr, 10);

        // Check for conversion errors
        if (endptr == argv[1] || *endptr != '\0' || errno == ERANGE) {
            fprintf(stderr, "Error: The provided argument is not a valid integer.\n");
            return 1;
        }
        
        int result = addThousand((int)inputVal);
        printf("%d\n", result);

    } else {
        printf("No command line argument provided. Running test cases:\n");
        // 5 test cases
        int testValues[] = {5, -10, 0, 999, -2000};
        int numTests = sizeof(testValues) / sizeof(testValues[0]);
        for (int i = 0; i < numTests; i++) {
            int val = testValues[i];
            int result = addThousand(val);
            printf("Input: %d, Output: %d\n", val, result);
        }
    }
    return 0;
}

File: benchmarkDS/ourMethod/GEMINI_gemini-2.5-pro/Lse21.py
import sys

def add_thousand(value_str):
    """
    Parses a string to an integer and adds 1000.
    
    Args:
        value_str: The string representation of the number.

    Returns:
        The integer result of the addition.

    Raises:
        ValueError: If the input string is not a valid integer.
    """
    try:
        # Python integers have arbitrary precision, so overflow on addition
        # is not a concern as it is in fixed-size integer languages.
        value = int(value_str)
        return value + 1000
    except ValueError:
        raise ValueError("Input must be a valid integer.")

def run_tests():
    """Runs a set of test cases."""
    print("--- Running Test Cases ---")
    test_cases = [
        "10",                      # Valid positive
        "0",                       # Valid zero
        "-50",                     # Valid negative
        "not a number",            # Invalid format
        "999999999999999999999999999999" # Large number
    ]
    for test in test_cases:
        try:
            result = add_thousand(test)
            print(f"Input: \"{test}\", Output: {result}")
        except ValueError as e:
            print(f"Input: \"{test}\", Error: {e}")

def main():
    """
    Main entry point of the script.
    Processes a command-line argument or runs tests.
    """
    if len(sys.argv) == 2:
        try:
            result = add_thousand(sys.argv[1])
            print(result)
            return 0  # Success
        except ValueError as e:
            print(f"Error: {e}", file=sys.stderr)
            return 1  # Failure
    elif len(sys.argv) == 1:
        run_tests()
        return 0 # Tests ran successfully
    else:
        print(f"Usage: python {sys.argv[0]} <integer>", file=sys.stderr)
        return 1 # Failure

if __name__ == "__main__":
    sys.exit(main())